crates/electrum/src/bdk_electrum_client.rs

```diff
@@ -527,10 +449,13 @@ fn construct_update_tip(
     let agreement_height = agreement_cp.as_ref().map(CheckPoint::height);
 
     let new_tip = new_blocks
-        .into_iter()
+        .iter()
         // Prune `new_blocks` to only include blocks that are actually new.
-        .filter(|(height, _)| Some(*height) > agreement_height)
-        .map(|(height, hash)| BlockId { height, hash })
+        .filter(|(height, _)| Some(*<&u32>::clone(height)) > agreement_height)
```

Comment:
nit, feel free to ignore

```suggestion
        .filter(|(&height, _)| Some(height) > agreement_height)
```